core/engine.go

```diff
@@ -147,6 +148,10 @@ func (e *Engine) StartOutputs() error {
 				})
 		}
 
+		if builtinMetricOut, ok := out.(output.WithBuiltinMetrics); ok {
+			builtinMetricOut.SetBuiltinMetrics(e.builtInMetrics)
```

Comment:
This seems unnecessary - you already do it in `StartOutputs()` in `cmd/run.go`

Reply:
oops this whole function should be dropped ;)